File: strong.cpp
#include<iostream>
using namespace std;
int factorial(int n)
{
	if(n == 0 || n == 1)
	{
		return 1;
	}
	return n*factorial(n-1);
}
bool isStrong(int n)
{
	int sum = 0;
	int num = n;
	while(num > 0)
	{
		sum += factorial(num%10);
		num /= 10;
	}
	return sum == n;
}
int main()
{
	int n;
	cout<<"Enter a number : ";
	cin>>n;
	
	if(isStrong(n))
	{
		cout<<"Is a Strong Number."<<endl;
	}
	else
	{
		cout<<"Not a Strong Number."<<endl;
	}
}

File: strong.py
def factorial(n:int)->int:
    if n == 0 or n == 1:
        return 1
    return n*factorial(n-1)

def isStrong(n:int)->bool:
    sum = 0
    num = n
    while(num > 0):
        sum += factorial(num%10)
        num //= 10
    return sum == n

if __name__ == "__main__":
    n = int(input("Enter a number : "))
    if(isStrong(n)):
        print("Is a Strong number.")
    else:
        print("Not a Strong number.")
    
